### PR TITLE
code style compatible with dotnet new version

### PR DESCRIPTION
**Pull request by mokarchi**
My changes include code style, I believe new changes related to newer versions of dotnet should be made in the code to make it compatible and more readable by new users

## Changes
Beginning with C# 9.0, constructor invocation expressions are target-typed. That is, if a target type of an expression is known, you can omit a type name
In C# 9 they added several new pattern matching operators that can be combined with the is operator
the null-coalescing assignment operator ??= assigns the value of its right-hand operand to its left-hand operand only if the left-hand operand evaluates to null and also new using declaration

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.AspNet/Implementation/HttpInListener.cs [hunk 1]
```diff
@@ -57,12 +57,7 @@ public void Dispose()
         /// <returns>Span uri value.</returns>
         private static string GetUriTagValueFromRequestUri(Uri uri)
         {
-            if (string.IsNullOrEmpty(uri.UserInfo))
-            {
-                return uri.ToString();
-            }
-
-            return string.Concat(uri.Scheme, Uri.SchemeDelimiter, uri.Authority, uri.PathAndQuery, uri.Fragment);
+            return string.IsNullOrEmpty(uri.UserInfo) ? uri.ToString() : string.Concat(uri.Scheme, Uri.SchemeDelimiter, uri.Authority, uri.PathAndQuery, uri.Fragment);
```
Comment (reyang) [inline src/OpenTelemetry.Instrumentation.AspNet/Implementation/HttpInListener.cs:60]: Would this line be too long? (e.g. the original version makes it easier to put a breakpoint on Ln62).

**Review by mic-max — APPROVED** [review]

**Comment by mokarchi** [comment]
I didn't no that why some check were not successful
@mic-max @reyang

**Comment by mokarchi** [comment]
@mic-max I don't know why codecov/project not successful.

**Comment by mic-max** [comment]
> @mic-max I don't know why codecov/project not successful.

That check isn't the best. Nothing to worry about on your end!

**Review by mic-max — APPROVED** [review]

**Pull request merged**